src/briefcase/commands/base.py

```diff
@@ -374,9 +375,18 @@ def download_url(self, url, download_path):
             )
 
         # The initial URL might (read: will) go through URL redirects, so
-        # we need the *final* URL name, from which we can extract the cache
-        # filename.
-        cache_name = urlparse(response.url).path.split('/')[-1]
+        # we need the *final* response. We look at either the `Content-Disposition`
+        # header, or the final URL, to extract the cache filename.
+        cache_full_name = urlparse(response.url).path
+        if 'Content-Disposition' in response.headers:
+            # See also https://tools.ietf.org/html/rfc6266
+            header_value = response.headers['Content-Disposition']
+            value, parameters = parse_header(header_value)
+            if value.split(':', 1)[-1].strip().lower() == 'attachment':
+                # We only the `filename=` parameter. RFC 6266 defines a more complicated
```

Comment:
We only.... what... the filename= parameters?

Reply:
+1 will fix :)